### PR TITLE
Made different implementations of client service proxies injectable

### PR DESCRIPTION
**Pull request by ahmetmircik**

### CODE REVIEW

File: hazelcast-client-new/src/main/java/com/hazelcast/client/impl/DefaultClientExtension.java [hunk 1]
```diff
@@ -93,4 +96,20 @@ public SocketInterceptor getSocketInterceptor() {
...
     }
+
+    /**
+     * Returns a proxy for the corresponding service.
+     *
+     * @param service service for the proxy to create.
+     * @return proxy for the service.
+     * @throws java.lang.IllegalArgumentException if type is not known
+     */
+    @Override
```
Comment (pveentjer) [inline hazelcast-client-new/src/main/java/com/hazelcast/client/impl/DefaultClientExtension.java]: Why do you copy paste the javadoc? The Javadoc tool automatically copied the javadoc of the method you implement/override. No need to copy paste it. Copy pasting leads to all kinds of problems, e.g. inconsistencies.

File: hazelcast-client/src/main/java/com/hazelcast/client/impl/DefaultClientExtension.java [hunk 2]
```diff
@@ -93,4 +96,21 @@ public SocketInterceptor getSocketInterceptor() {
...
+
+    /**
+     * Returns a proxy for the corresponding service.
+     *
+     * @param service service for the proxy to create.
+     * @return proxy for the service.
+     * @throws java.lang.IllegalArgumentException if type is not known
+     */
+    @Override
+    public <T> Class<? extends ClientProxy> getServiceProxy(Class<T> service) {
```
Comment (pveentjer) [inline hazelcast-client/src/main/java/com/hazelcast/client/impl/DefaultClientExtension.java]: No copy/paste of javadoc is needed.

**Comment by pveentjer** [comment]
New methods, but no new tests. Can you add unit tests for the methods you have created.

**Comment by ahmetmircik** [comment]
done

**Comment by pveentjer** [comment]
Thanks. Will merge once green.

**Pull request merged**